engine/consensus/matching/engine_test.go

```diff
@@ -57,421 +52,187 @@ func TestMatchingEngine(t *testing.T) {
 		log:                     log,
 		engineMetrics:           metrics,
 		mempool:                 metrics,
 		metrics:                 metrics,
-		state:                   ms.state,
+		state:                   ms.State,
 		requester:               ms.requester,
-		resultsDB:               ms.resultsDB,
-		headersDB:               ms.headersDB,
-		indexDB:                 ms.indexDB,
-		incorporatedResults:     ms.resultsPL,
-		approvals:               ms.approvalsPL,
-		seals:                   ms.sealsPL,
+		resultsDB:               ms.ResultsDB,
+		headersDB:               ms.HeadersDB,
+		indexDB:                 ms.IndexDB,
+		incorporatedResults:     ms.ResultsPL,
+		receipts:                ms.ReceiptsPL,
+		approvals:               ms.ApprovalsPL,
+		seals:                   ms.SealsPL,
 		checkingSealing:         atomic.NewBool(false),
 		requestReceiptThreshold: 10,
 		maxResultsToRequest:     200,
-		assigner:                ms.assigner,
+		assigner:                ms.Assigner,
 		requireApprovals:        true,
+		receiptValidator:        ms.receiptValidator,
 	}
 }
 
-func (ms *MatchingSuite) TestOnReceiptInvalidOrigin() {
-	// we don't validate the origin of an execution receipt anymore, as Execution Nodes
-	// might forward us Execution Receipts from others for blocks they haven't computed themselves
-	ms.T().Skip()
-
-	// try to submit a receipt with a random origin ID
-	originID := ms.exeID
-	receipt := unittest.ExecutionReceiptFixture()
-
-	err := ms.matching.onReceipt(originID, receipt)
-	ms.Require().Error(err, "should reject receipt with mismatching origin and executor")
-
-	ms.resultsDB.AssertNumberOfCalls(ms.T(), "Store", 0)
-	ms.resultsPL.AssertNumberOfCalls(ms.T(), "Add", 0)
-	ms.sealsPL.AssertNumberOfCalls(ms.T(), "Add", 0)
-}
-
-// try to submit a receipt from a NON-ExecutionNode  (here: consensus node)
+// Test that we reject receipts for unknown blocks without generating an error
 func (ms *MatchingSuite) TestOnReceiptUnknownBlock() {
-	originID := ms.conID
-	receipt := unittest.ExecutionReceiptFixture(unittest.WithExecutorID(originID))
+	// This receipt has a random block ID, so the matching engine won't find it.
+	receipt := unittest.ExecutionReceiptFixture()
 
-	// onReceipt should not throw an error
-	err := ms.matching.onReceipt(originID, receipt)
-	ms.Require().NoError(err, "should ignore receipt for unknown block")
+	// onReceipt should reject the receipt without throwing an error
+	err := ms.matching.onReceipt(receipt.ExecutorID, receipt)
+	ms.Require().NoError(err, "should drop receipt for unknown block without error")
 
-	ms.resultsDB.AssertNumberOfCalls(ms.T(), "Store", 0)
-	ms.resultsPL.AssertNumberOfCalls(ms.T(), "Add", 0)
-	ms.sealsPL.AssertNumberOfCalls(ms.T(), "Add", 0)
+	ms.ReceiptsPL.AssertNumberOfCalls(ms.T(), "Add", 0)
+	ms.ResultsPL.AssertNumberOfCalls(ms.T(), "Add", 0)
 }
 
-// try to submit a receipt for a known block from a consensus node
-func (ms *MatchingSuite) TestOnReceiptInvalidRole() {
```

Comment:
checking Receipt validity is now handled by `ReceiptValidator`. For the matching engine, we only need to test that the output of `ReceiptValidator` is processed properly: see test https://github.com/onflow/flow-go/blob/32081ff9f91d24fd5c10ab57bdf447a70c7dd4ef/engine/consensus/matching/engine_test.go#L212-L213